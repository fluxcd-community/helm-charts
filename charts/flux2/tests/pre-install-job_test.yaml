suite: test pre-install-job hook
templates:
  - pre-install-job.yaml
tests:
  - it: should have no annotations added by default
    asserts:
      - isEmpty:
          path: spec.template.metadata.annotations
  - it: should have added annotation
    set:
      cli:
        annotations:
          name1: value1
    asserts:
      - isNotEmpty:
          path: spec.template.metadata.annotations
      - equal:
          path: spec.template.metadata.annotations
          value:
            name1: value1
  - it: should have added annotations
    set:
      cli:
        annotations:
          name1: value1
          name2: value2
          name3: value3
    asserts:
      - isNotEmpty:
          path: spec.template.metadata.annotations
      - equal:
          path: spec.template.metadata.annotations
          value:
            name1: value1
            name2: value2
            name3: value3
  - it: should set automount service account token
    set:
      cli.serviceAccount.automount: false
    asserts:
      - equal:
          path: spec.template.spec.automountServiceAccountToken
          value: false
  - it: should set resource requests and limits
    set:
      cli.resources:
        limits:
          cpu: 300m
          memory: 1024Mi
        requests:
          cpu: 300m
          memory: 1024Mi
    asserts:
      - equal:
          path: spec.template.spec.containers[0].resources.limits.cpu
          value: 300m
      - equal:
          path: spec.template.spec.containers[0].resources.limits.memory
          value: 1024Mi
      - equal:
          path: spec.template.spec.containers[0].resources.requests.cpu
          value: 300m
      - equal:
          path: spec.template.spec.containers[0].resources.requests.memory
          value: 1024Mi
  - it: should override securityContext
    set:
      cli.securityContext:
        runAsNonRoot: false
    asserts:
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            runAsNonRoot: false
  - it: should set volumes
    set:
      cli.volumeMounts:
        - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
          name: serviceaccount-token
      cli.volumes:
        - name: serviceaccount-token
          projected:
            defaultMode: 420
            sources:
            - serviceAccountToken:
                expirationSeconds: 3607
                path: token
            - configMap:
                name: kube-root-ca.crt
                items:
                - key: ca.crt
                  path: ca.crt
            - downwardAPI:
                items:
                - path: namespace
                  fieldRef:
                    apiVersion: v1
                    fieldPath: metadata.namespace
    asserts:
      - isNotEmpty:
          path: spec.template.spec.volumes
      - isNotEmpty:
          path: spec.template.spec.containers[0].volumeMounts
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
