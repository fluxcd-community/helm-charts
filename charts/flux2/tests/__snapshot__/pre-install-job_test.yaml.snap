should match snapshot of default values:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        helm.sh/hook: pre-install
        helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
        helm.sh/hook-weight: "-5"
      labels:
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        app.kubernetes.io/version: 0.37.0
        helm.sh/chart: flux2-2.5.1
      name: RELEASE-NAME-flux-check
    spec:
      backoffLimit: 1
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: NAMESPACE
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/part-of: flux
            app.kubernetes.io/version: 0.37.0
            helm.sh/chart: flux2-2.5.1
          name: RELEASE-NAME
        spec:
          automountServiceAccountToken: true
          containers:
          - command:
            - /usr/local/bin/flux
            - check
            - --pre
            - --namespace
            - NAMESPACE
            image: ghcr.io/fluxcd/flux-cli:v0.37.0
            name: flux-cli
            securityContext:
              allowPrivilegeEscalation: false
              capabilities:
                drop:
                - ALL
              readOnlyRootFilesystem: true
              runAsNonRoot: true
              seccompProfile:
                type: RuntimeDefault
          restartPolicy: Never
          serviceAccountName: RELEASE-NAME-flux-check
